Troubleshooting instructions for clidriver errror (#15882)
DataDog/integrations-core: ibm_db2/README.md

## Troubleshooting

### CLI Driver SQL1531N error

> If you encounter an issue like the following:

Reviewer: ```suggestion
If you encounter an issue that produces error logs like the following:
```

```
2023-08-10 23:34:47 UTC | CORE | ERROR | (pkg/collector/python/datadog_agent.go:129 in LogMessage) | ibm_db2:c051131490335a94 | (ibm_db2.py:563) | Unable to connect to database `datadog` as user `db2inst1`: [IBM][CLI Driver] SQL1531N  The connection failed because the name specified with the DSN connection string keyword could not be found in either the db2dsdriver.cfg configuration file or the db2cli.ini configuration file.  Data source name specified in the connection string: "DATADOG". SQLCODE=-1531
```

> Then it's most likely one of the following scenarios:

Reviewer: ```suggestion
Then it's most likely caused by one of the following scenarios:
```

> - The configuration (conf.yaml) is missing a host and port configuration

Reviewer 2: I see that host is optional but it does not default to localhost. What happens when no host is provided?

> - The CLI Driver isn't able to locate the database due to the absence of db2cli.ini and db2dsdriver.cfg

Reviewer: ```suggestion
- The CLI Driver isn't able to locate the database due to the absence of `db2cli.ini` and `db2dsdriver.cfg`
```

> The agent requires the information in both of the above scenarios to figure out where to properly connect to the database. To solve this issue, you can either include a host and port parameter for every instance of the ibm_db2 check experiencing this issue, or, alternatively, if you would like to use the DSNs defined in either the db2cli.ini or db2dsdriver.cfg files, you can copy those files over to the clidriver directory that the agent uses. Under normal circumstances, that directory will be located at /opt/datadog-agent/embedded/lib/python3.9/site-packages/clidriver/cfg for Linux.

Reviewer: ```suggestion
The Agent requires the information in both of the above scenarios to determine where to properly connect to the database. To solve this issue, you can either include a host and port parameter for every instance of the `ibm_db2` check experiencing this issue, or, alternatively, if you want to use the DSNs defined in either the `db2cli.ini` or `db2dsdriver.cfg` files, you can copy those files over to the `clidriver` directory that the Agent uses. Under normal circumstances, that directory is located at `/opt/datadog-agent/embedded/lib/python3.9/site-packages/clidriver/cfg` for Linux.
```

### CLI Driver SQL1531N error

If you encounter an issue that produces error logs like the following:

```
2023-08-10 23:34:47 UTC | CORE | ERROR | (pkg/collector/python/datadog_agent.go:129 in LogMessage) | ibm_db2:c051131490335a94 | (ibm_db2.py:563) | Unable to connect to database `datadog` as user `db2inst1`: [IBM][CLI Driver] SQL1531N  The connection failed because the name specified with the DSN connection string keyword could not be found in either the db2dsdriver.cfg configuration file or the db2cli.ini configuration file.  Data source name specified in the connection string: "DATADOG". SQLCODE=-1531
```

Then it's most likely caused by one of the following scenarios:
- The configuration (conf.yaml) is missing a host and port configuration
- The CLI Driver isn't able to locate the database due to the absence of `db2cli.ini` and `db2dsdriver.cfg`

> The Agent requires the information in both of the above scenarios to determine where to properly connect to the database. To solve this issue, you can either include a host and port parameter for every instance of the `ibm_db2` check experiencing this issue, or, alternatively, if you want to use the DSNs defined in either the `db2cli.ini` or `db2dsdriver.cfg` files, you can copy those files over to the `clidriver` directory that the Agent uses. Under normal circumstances, that directory is located at `/opt/datadog-agent/embedded/lib/python3.9/site-packages/clidriver/cfg` for Linux.

Reviewer: ```suggestion
The Agent requires the information in both of the above scenarios to determine where to properly connect to the database. To solve this issue, you can either include a host and port parameter for every instance of the `ibm_db2` check experiencing this issue. Alternatively, if you want to use the DSNs defined in either the `db2cli.ini` or `db2dsdriver.cfg` files, you can copy those files over to the `clidriver` directory that the Agent uses. Under normal circumstances, that directory is located at `/opt/datadog-agent/embedded/lib/python3.9/site-packages/clidriver/cfg` for Linux.
```
